### PR TITLE
Move distributor's validations into distributor package

### PR DESCRIPTION
**Pull request by duricanikolic**
#### What this PR does

This PR is another refactoring related to the improved error handling of distributor errors.
All distributor-related validations have been moved from the `validation` to the `distributor` package. Definitions of the errors introduced in https://github.com/grafana/mimir/pull/6191 and signatures of their constructors have been updated accordingly.

#### Which issue(s) this PR fixes or relates to

Related to https://github.com/grafana/mimir/issues/1900, https://github.com/grafana/mimir/issues/6008

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/distributor/validate.go [hunk 1]
```diff
@@ -355,9 +353,9 @@ func (m *MetadataValidationMetrics) DeleteUserMetrics(userID string) {
 
 func NewMetadataValidationMetrics(r prometheus.Registerer) *MetadataValidationMetrics {
```
Comment (colega) [inline pkg/distributor/validate.go:354]: `MetadataValidationMetrics` and similar don't need to be exported anymore.

Comment (duricanikolic) [reply pkg/distributor/validate.go]: I have fixed that. Thank you.

File: pkg/util/discarded_metrics.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```
Comment (duricanikolic) [inline pkg/util/discarded_metrics.go:1]: Note to reviewers: these methods used to live in the `validation` package, but they are called only from the `distributor` package (all of them) and from the `ingester` package (`DiscardedSamplesCounter()` only). Therefore I put them in a separate util source.

**Review by pstibrany — APPROVED** [review]
Changes look good to me, thank you.

**Pull request merged**